app/models/manageiq/providers/ibm_cloud/inventory/parser/power_virtual_servers.rb

```diff
@@ -197,4 +199,22 @@ def sshkeys
       persister.auth_key_pairs.build(:name => tenant_key[:name])
     end
   end
+
+  def systemtypes
+    systype_lst = []
+    collector.system_pool.values.each { |v| systype_lst |= [v['type']] }
```

Comment:
You can do this with `.map` which builds an array out of each value returned by the block so e.g. `collector.system_pool.values.map { |v| v["type"] }`